### DISCUSSION TITLE
LoRaWan f_cnt value

### DISCUSSION THREAD

**Discussion by michapr** [jgromes/RadioLib#971] [General]
An unimportant detail, but maybe it could be corrected?

In TTN I see the f_cnt value is different from used node.getFcntUp().
I'm sending this value to TTN, that's why have seen it.

The value in the TTN (f_cnt) is always one higher than in the software.
Not sure if course,.. maybe it is correct?

Certainly it is completely unimportant and insignificant... :))

**Comment by StevenCellist** [comment]
https://github.com/jgromes/RadioLib/blob/a907026c7b2a213e62a631c092b32a273616ce6a/examples/LoRaWAN/LoRaWAN_End_Device_Reference/LoRaWAN_End_Device_Reference.ino#L169C1-L173C52
If you check the comment, you'll see that the FCnt in the code is from the _last_ uplink, which is retrieved _before_ sending a new uplink. So that means that the value in the payload is always one lower indeed. If you were to print out the result of `node.getFcntUp()` after sending the uplink, I'd expect it would match.

**Reply by michapr** [reply]
Great!

Thanks, I really didn't notice the comment there, I'm sorry for that ... :(
